### PR TITLE
Add endpoints for RestaurantServlet as well as servlet tests

### PR DESCRIPTION
**Pull request by lisasgoh**

### CODE REVIEW

File: src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java [hunk 1]
```diff
@@ -63,7 +63,9 @@ public Restaurant updateRestaurant(Restaurant restaurant) {
       restaurantEntity.setProperty("photoBlobkey", restaurant.photoBlobkey);
     }
     datastore.put(restaurantEntity);
-    return readRestaurant(restaurant.id);
+    String name = (String) restaurantEntity.getProperty("name");
+    String photoBlobkey = (String) restaurantEntity.getProperty("photoBlobkey");
+    return new Restaurant(restaurant.id, name, photoBlobkey);
```
Comment (limli) [inline src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java]: move these changes to PR #31

File: src/main/java/com/google/step/servlets/RestaurantServlet.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    String name = request.getParameter("name");
+    String photoBlobkey = "A_BLOB_KEY"; // TODO Blobkey
+    Restaurant restaurant = new Restaurant(id, name, photoBlobkey);
+    Restaurant updatedRestaurant = manager.updateRestaurant(restaurant);
+    if (updatedRestaurant == null) {
+      response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+    } else {
+      response.getWriter().println(JsonFormatter.getRestaurantJson(updatedRestaurant));
+    }
+    response.sendRedirect("/restaurant/" + id);
```
Comment (limli) [inline src/main/java/com/google/step/servlets/RestaurantServlet.java]: you want to write or redirect? If you redirect the page, then you can't see what the writer wrote

**Review by limli — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java (hunk 1 above)
Comment (lisasgoh) [reply src/main/java/com/google/step/datamanager/RestaurantManagerDatastore.java]: i changed it back to readrestaurant which is the same as the other PR

File: src/main/java/com/google/step/servlets/RestaurantServlet.java (hunk 2 above)
Comment (lisasgoh) [reply src/main/java/com/google/step/servlets/RestaurantServlet.java]: removed redirect

File: src/main/java/com/google/step/servlets/RestaurantServlet.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles individual restaurants */
+@WebServlet("/api/restaurant/*")
+public class RestaurantServlet extends HttpServlet {
+
+  private RestaurantManager manager = new RestaurantManagerDatastore();
+
+  public RestaurantServlet(RestaurantManager restaurantManager) {
```
Comment (limli) [inline src/main/java/com/google/step/servlets/RestaurantServlet.java:18]: missing constructor without arguments. You should run the app and go to the backend url manually to test on top of running `mvn test`.

**Review by limli — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/servlets/RestaurantServlet.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import com.google.step.datamanager.RestaurantManagerDatastore;
+import com.google.step.model.Restaurant;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles individual restaurants */
+@WebServlet("/api/restaurant/*")
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/servlets/RestaurantServlet.java]: `/api/restaurants/*`

File: src/main/java/com/google/step/servlets/RestaurantPostServlet.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import com.google.step.datamanager.RestaurantManagerDatastore;
+import com.google.step.model.Restaurant;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles posting restaurants. */
+@WebServlet("/api/restaurant")
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/servlets/RestaurantPostServlet.java]: `/api/restaurants`

File: src/main/java/com/google/step/servlets/RestaurantServlet.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    long id;
+    try {
+      id = Long.parseLong(request.getPathInfo().substring(1));
+    } catch (NumberFormatException e) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    } catch (StringIndexOutOfBoundsException e) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/servlets/RestaurantServlet.java]:
```suggestion
    try {
      id = Long.parseLong(request.getPathInfo().substring(1));
    } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
      return;
    }
```

File: src/main/java/com/google/step/servlets/RestaurantServlet.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
```
Comment (anuragshirolkar) [inline src/main/java/com/google/step/servlets/RestaurantServlet.java]: same as above

File: src/test/java/com/google/step/servlets/RestaurantServletTest.java [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+  private static final String UPDATE_NAME_A = "UPDATE";
+  private static final Restaurant UPDATE_RESTAURANT_A =
+      new Restaurant(ID_A, UPDATE_NAME_A, BLOBKEY_A);
+
+  private RestaurantManager restaurantManager;
+
+  private RestaurantServlet restaurantServlet;
+
+  private RestaurantPostServlet restaurantPostServlet;
```
Comment (anuragshirolkar) [inline src/test/java/com/google/step/servlets/RestaurantServletTest.java]: Separate these out into 2 different test classes.

**Review by anuragshirolkar — CHANGES_REQUESTED** [review]

**Review by limli — APPROVED** [review]

**Review by anuragshirolkar — APPROVED** [review]

**Pull request merged**